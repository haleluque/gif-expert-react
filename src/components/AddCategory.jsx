import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onSetCategory}) => {

    const [inputValue, setInputValue] = useState("");
    
    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        //previene que se haga un refresh sobre la pagina cuando se dispara el evento
        event.preventDefault();
        if(inputValue.trim().length <= 0) return;
        onSetCategory(inputValue);
        setInputValue("");
    }

    //aria-label necesario para que se pueda buscar el formulario en los test (by role)
    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text"
                placeholder="Buscar Gifs"
                onChange={ onInputChange }
                value={ inputValue ?? '' }
            />
        </form>
    )
}

AddCategory.prototypes = {
    onSetCategory : PropTypes.func.isRequired
}
