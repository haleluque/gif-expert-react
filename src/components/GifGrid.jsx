import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({category}) => {
    //Se creo un custom hook para administrar los gifs
    const {images, isLoading} = useFetchGifs(category);
  
    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            
            <div className=".card-grid">
                {
                    //spread operator se usa para enviar cada una de las propiedades del objeto como props al componente
                    images.map((image) => (
                        <GifItem 
                            key={image.id} 
                            { ...image } />
                    ))
                }
            </div>
        </>
    )
}