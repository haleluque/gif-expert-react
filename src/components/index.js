/**
 * Esto es un archivo de barril, usado para juntar referencias de archivos en un solo lugar
 * y de esa forma poder hacer menos imports en los archivos de componentes
 */
export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';