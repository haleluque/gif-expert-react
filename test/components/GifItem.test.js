import { GifItem } from "../../src/components/GifItem";
import { render, screen } from "@testing-library/react"

describe('Pruebas componenete GifItem', () => { 
    const title = "Saitama";
    const url = "https://www.url.com/";
    
    test('Debe hacer match con el snapshot', () => { 
        const {container} = render( <GifItem title={title} url={url} /> )
        expect(container).toMatchSnapshot();
     });

    test('Debe mostrar la imagen con el url y alt indicado', () => { 
        render( <GifItem title={title} url={url} /> )
        //screen.debug();
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
     });

    test('El titulo debe mostrarse como texto', () => { 
        render( <GifItem title={title} url={url} /> )
        expect(screen.getByText(title)).toBeTruthy();
    });
 })