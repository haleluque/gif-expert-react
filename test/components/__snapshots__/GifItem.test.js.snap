// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componenete GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://www.url.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
